File: Jay/lv1/142086. 가장 가까운 같은 글자 /가장 가까운 같은 글자.swift
import Foundation

func solution(_ s:String) -> [Int] {
    var dic: [Character: Int] = [:]
    var array: [Int] = []
    
    for (index, char) in s.enumerated() {
        if let lastIndex = dic[char] {
            array.append(index - lastIndex)
        } else {
            array.append(-1)
        }
        dic[char] = index
    }
    return array
}

File: Jay/lv1/142086. 가장 가까운 같은 글자 /가장 가까운 같은 글자.kt
class Solution {
    fun solution(s: String): IntArray {
        val dic = mutableMapOf<Char, Int>()
        val array = mutableListOf<Int>()
    
        s.forEachIndexed { index, char ->
            val lastIndex = dic[char]
            if (lastIndex != null) {
                array.add(index - lastIndex)
            } else {
                array.add(-1)
            }
            dic[char] = index
    }
    return array.toIntArray()
    }
}
